tests/performance/avg_serialization.xml

```diff
@@ -1,7 +1,7 @@
 <test>
     <create_query>DROP TABLE IF EXISTS test_avg_insert</create_query>
     <create_query>
-        CREATE TABLE test_avg_insert (key UInt64, value AggregateFunction(avg, UInt8)) ENGINE = Memory()
+        CREATE TABLE test_avg_insert (key UInt64, value AggregateFunction(avg, UInt8)) ENGINE = Memory() SETTINGS max_rows_to_keep=1
```

Comment:
Will this help if we have only one INSERT? `max_rows_to_keep` can only remove the whole `block`, so it will be no-op for this case - https://github.com/ClickHouse/ClickHouse/blob/5d5f53e1628fb2efc83b3e7aef578fd43369bd81/src/Storages/StorageMemory.cpp#L123-L134

And if we will simply add `max_block_size=1` into `INSERT` this will became too slow, maybe we just need to replace `Memory` with i.e. `TinyLog`? Or try `compress=true`, although unlikely `AggregateFunction` can be compressed significantly

Reply:
>TinyLog is much slower and does not show the improvement quite so obviously. With memory the improvement of the original PR is ~8x, while TinyLog is 1.3x as most of the time is spent writing to disk.

Maybe we can use File over /dev/null where we don't need the data

Reply:
Using `Memory() SETTINGS max_rows_to_keep=1` is 4x faster than `File('Native', '/dev/null')` and 9x faster than `Null`. Why? I don't know, it doesn't really make any sense.